src/main/java/io/odpf/firehose/factory/FirehoseConsumerFactory.java

```diff
@@ -185,18 +186,17 @@ private Sink withRetry(Sink basicSink, GenericKafkaFactory genericKafkaFactory,
                 new Instrumentation(statsDReporter, ExponentialBackOffProvider.class),
                 new BackOff(new Instrumentation(statsDReporter, BackOff.class)));
 
-        if (appConfig.isDlqEnable()) {
+        if (!appConfig.isDlqEnable() || appConfig.getSinkType() == SinkType.OBJECTSTORAGE) {
```

Comment:
why one specific check for a sink? Every sink should have the same behaviour!

Reply:
will need to discuss this, currently `ObjectStorageSink` uses different DLQ approach compared to other Sink

Reply:
yeah. we should think about solution in general. dlq should work for every sink, if the configs are set properly OR correct methods are implemented. There is nothing special about ObjectStorageSInk

Reply:
removed this too